### PR TITLE
#166789870 Descriptive signup validation errors

### PR DESCRIPTION
**Pull request by Dom58**
#### What does this PR do?
This pull request:
- create a middle-wares subdirectory to keep the signup validator
- create a signup validator
- unit test of signup validation branch
- edit user route by adding a signup validator as a middleware

#### Description of Task to be completed?
- Users should receive descriptive signup validation errors
#### How should this be manually tested?
- First clone it to your machine:
```
https://github.com/andela/ah-lobos-backend.git
```
- Open it using your favorite IDE, I recommend ([vs code](https://code.visualstudio.com/download))

- Install all necessary node modules
```
npm install
```
- Run
```
npm run dev
```
Remote testing via [heroku](https://ah-lobos-backend-swagger.herokuapp.com)
#### Any background context you want to provide?
- Create `.env` file with the help of `example.env`
- Access the user sign-up route via /api/users on the HTTP POST method and interact with signup body
#### What are the relevant pivotal tracker stories?
[#166789870](https://www.pivotaltracker.com/story/show/166789870)
#### Screenshots
Screen Shot 2019-07-02 at 19 55 25

- signup validation test 
Screen Shot 2019-07-02 at 20 44 33

- Swagger documentation
Screen Shot 2019-07-05 at 17 12 51

#### Questions:
N/A

### CODE REVIEW

**Review by Emile-Nsengimana — COMMENTED** [review]
What do you mean with ``` test validation branch ```?

**Review by j4l13n — APPROVED** [review]

**Comment by Dom58** [comment]
> What do you mean with `test validation branch`?

I guess you was asking about the PR description, right? If so, I was talking about unit test of  `ft-signup-validation-erros-166789870` branch

**Review by adafia — APPROVED** [review]

**Review by manzif — APPROVED** [review]

**Review by manzif — APPROVED** [review]

**Review by Emile-Nsengimana — APPROVED** [review]

File: src/middlewares/signup.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const user = {
+      username, email, password
+    };
+    const checkUser = Joi.validate(user, signupSchema, {
+      abortEarly: false
+    });
+    if (checkUser.error) {
+      const Errors = [];
+      // eslint-disable-next-line no-plusplus
+      for (let i = 0; i < checkUser.error.details.length; i++) {
```
Comment (emukungu) [inline src/middlewares/signup.js]: instead of ignoring code using `eslint-disable-next-line no-plusplus`, use the proper operators

Comment (Dom58) [reply src/middlewares/signup.js]: Thank you let's change it!

**Pull request merged**